pkg/controller/pd_control.go

```diff
@@ -20,15 +20,19 @@ import (
 
 // getPDClientFromService gets the pd client from the TidbCluster
 func getPDClientFromService(pdControl pdapi.PDControlInterface, tc *v1alpha1.TidbCluster) pdapi.PDClient {
-	if tc.IsHeterogeneous() {
+	if tc.HeterogeneousWithoutLocalPD() {
 		if len(tc.Spec.ClusterDomain) > 0 {
-			return pdControl.GetClusterRefPDClient(pdapi.Namespace(tc.GetNamespace()), tc.Spec.Cluster.Name, tc.Spec.ClusterDomain, tc.IsTLSClusterEnabled())
+			// FIXME: across k8s cluster without local pd
+			// if tls enable tc.Spec.Cluster.Name should be same as tc.Name? because it will query the secret using this name?
+			return pdControl.GetClusterRefPDClient(pdapi.Namespace(tc.Spec.Cluster.Namespace), tc.Spec.Cluster.Name, tc.Spec.Cluster.ClusterDomain, tc.IsTLSClusterEnabled())
 		}
-		return pdControl.GetPDClient(pdapi.Namespace(tc.GetNamespace()), tc.Spec.Cluster.Name, tc.IsTLSClusterEnabled())
+		return pdControl.GetPDClient(pdapi.Namespace(tc.Spec.Cluster.Namespace), tc.Spec.Cluster.Name, tc.IsTLSClusterEnabled())
 	}
+
 	if len(tc.Spec.ClusterDomain) > 0 {
 		return pdControl.GetClusterRefPDClient(pdapi.Namespace(tc.GetNamespace()), tc.GetName(), tc.Spec.ClusterDomain, tc.IsTLSClusterEnabled())
```

Comment:
I think we should use `GetPDClient` here, WDYT @july2993

Reply:
https://github.com/pingcap/tidb-operator/pull/3947/commits/c407ccbef8ad43e1138125f35e355f79545380fb